x-pack/platform/plugins/shared/streams_app/public/components/stream_list_view/documents_column.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiI18nNumber,
+  EuiSkeletonRectangle,
+  EuiDelayRender,
+} from '@elastic/eui';
+import { euiThemeVars } from '@kbn/ui-theme';
+import { css } from '@emotion/css';
+import moment from 'moment';
+import {
+  BarSeries,
+  Chart,
+  ScaleType,
+  Settings,
+  LIGHT_THEME,
+  DARK_THEME,
+  niceTimeFormatter,
+  Tooltip,
+  TooltipStickTo,
+  type SettingsProps,
+} from '@elastic/charts';
+import { i18n } from '@kbn/i18n';
+import { useEuiTheme } from '@elastic/eui';
+import { TimefilterHook } from '@kbn/data-plugin/public/query/timefilter/use_timefilter';
+import { useStreamsAppFetch } from '../../hooks/use_streams_app_fetch';
+import { useKibana } from '../../hooks/use_kibana';
+import { esqlResultToTimeseries } from '../../util/esql_result_to_timeseries';
+
+export function DocumentsColumn({
+  timefilter,
+  indexPattern,
+  numDataPoints,
+}: {
+  timefilter: TimefilterHook; // Workaround to keep state in sync
```

Comment:
why this workaround? I guess alternatively we could call the hook in this compoent itself, right?

Reply:
We can't use the `useTimerange` hook more than once on the page since it uses local state and we need the time range to be in sync across the entire page. 

I would suggest to create a Provider/hook that keeps the state in global context.

Reply:
They should all synchronize through the global `timefilter` service which is part of the data plugin (this is also what retains the current time range when moving from page to page within the streams UI):

https://github.com/elastic/kibana/blob/86a71033b3d81687d8ae3a07e654d12c5d757f17/src/platform/plugins/shared/data/public/query/timefilter/use_timefilter.ts#L53

I'm not sure about the overhead though, a provider would work too

Reply:
They are only synchronised at the time filter service level. 
The absolute time is set by the hook using local state so that is not in sync between different instances of the hook.

Reply:
I see, the absolute time range would have small differences across the different visualizations, that is indeed weird. Passing it in works fine I guess - ideally the absolute time range should be controlled centrally in the service as well, but oh well.